PWGLF/Tasks/LFNucleiBATask.cxx

```diff
@@ -216,33 +234,57 @@ struct LFNucleiBATask {
+      } else {
+        histos.fill(HIST("tracks/proton/h2antiProtonVspTNSigmaTPC"), track.pt(), track.nsigTPCPr());
+        histos.fill(HIST("tracks/deuteron/h2antiDeuteronVspTNSigmaTPC"), track.pt(), track.nsigTPCD());
+        histos.fill(HIST("tracks/helium/h2antiHeliumVspTNSigmaTPC"), track.pt(), track.nsigTPC3He());
+      }
       //  TOF
       histos.fill(HIST("tracks/pion/h2PionVspTNSigmaTOF"), track.pt(), track.nsigTOFPi());
       histos.fill(HIST("tracks/kaon/h2KaonVspTNSigmaTOF"), track.pt(), track.nsigTOFKa());
-      histos.fill(HIST("tracks/proton/h2ProtonVspTNSigmaTOF"), track.pt(), track.nsigTOFPr());
-      histos.fill(HIST("tracks/deuteron/h2DeuteronVspTNSigmaTOF"), track.pt(), track.nsigTOFD());
-      histos.fill(HIST("tracks/helium/h2HeliumVspTNSigmaTOF"), track.pt(), track.nsigTOF3He());
+      if (track.sign() > 0) {
+        histos.fill(HIST("tracks/proton/h2ProtonVspTNSigmaTOF"), track.pt(), track.nsigTOFPr());
+        histos.fill(HIST("tracks/deuteron/h2DeuteronVspTNSigmaTOF"), track.pt(), track.nsigTOFD());
+        histos.fill(HIST("tracks/helium/h2HeliumVspTNSigmaTOF"), track.pt(), track.nsigTOF3He());
+      } else {
+        histos.fill(HIST("tracks/proton/h2antiProtonVspTNSigmaTOF"), track.pt(), track.nsigTOFPr());
+        histos.fill(HIST("tracks/deuteron/h2antiDeuteronVspTNSigmaTOF"), track.pt(), track.nsigTOFD());
+        histos.fill(HIST("tracks/helium/h2antiHeliumVspTNSigmaTOF"), track.pt(), track.nsigTOF3He());
+      }
 
       // PID
-      if (std::abs(track.nsigTPCPr()) < nsigmaTPCcut)
-        histos.fill(HIST("tracks/proton/h1ProtonSpectra"), track.pt());
+      if (std::abs(track.nsigTPCPr()) < nsigmaTPCcut) {
+        if (track.sign() > 0)
+          histos.fill(HIST("tracks/proton/h1ProtonSpectra"), track.pt());
+        else
+          histos.fill(HIST("tracks/proton/h1antiProtonSpectra"), track.pt());
+      }
 
-      if (std::abs(track.nsigTPCD()) < nsigmaTPCcut)
-        histos.fill(HIST("tracks/deuteron/h1DeuteronSpectra"), track.pt());
+      if (std::abs(track.nsigTPCD()) < nsigmaTPCcut) {
+        if (track.sign() > 0)
+          histos.fill(HIST("tracks/deuteron/h1DeuteronSpectra"), track.pt());
+        else
+          histos.fill(HIST("tracks/deuteron/h1antiDeuteronSpectra"), track.pt());
+      }
 
-      if (std::abs(track.nsigTPC3He()) < nsigmaTPCcut)
-        histos.fill(HIST("tracks/helium/h1HeliumSpectra"), track.pt());
+      if (std::abs(track.nsigTPC3He()) < nsigmaTPCcut) {
+        if (track.sign() > 0)
+          histos.fill(HIST("tracks/helium/h1HeliumSpectra"), track.pt());
+        else
+          histos.fill(HIST("tracks/helium/h1antiHeliumSpectra"), track.pt());
+      }
 
       if (track.hasTOF()) {
-        histos.fill(HIST("tracks/h2TOFbetaVsP"), track.p(), track.beta());
+        histos.fill(HIST("tracks/h2TOFbetaVsP"), track.p() / (1.f * track.sign()), track.beta());
```

Comment:
I think sign is just a `short` equal to `1` or `-1` this is ok but you could also consider `track.p() * track.sign()` as it is not the charge, it is really the sign